ethcore/src/account.rs

```diff
@@ -16,24 +16,27 @@
 
 //! Single account in the system.
 
+use std::mem::replace;
 use std::collections::hash_map::Entry;
 use util::*;
 use pod_account::*;
 use account_db::*;
+use lru_cache::LruCache;
 
-use std::cell::{Ref, RefCell, Cell};
+use std::cell::{RefCell, Cell};
 
 /// Single account in the system.
-#[derive(Clone)]
 pub struct Account {
 	// Balance of the account.
 	balance: U256,
 	// Nonce of the account.
 	nonce: U256,
 	// Trie-backed storage.
 	storage_root: H256,
-	// Overlay on trie-backed storage - tuple is (<clean>, <value>).
-	storage_overlay: RefCell<HashMap<H256, (Filth, H256)>>,
+	// Cache of trie-backed storage
+	storage_cache: RefCell<LruCache<H256, H256>>,
```

Comment:
Should be made explicit in the struct docs about the meaning (and precedence) of these members. Particularly what happens when the same key appears in both.